tests/CodeBehind/UnitTests/BuildTests.cs

```diff
@@ -594,35 +619,43 @@ string Regexify (string input)
 				.Replace ("}", "\\}");
 		}
 
-		[OneTimeSetUp]
-		public void BuildTestsSetUp ()
+		string PrepareProject (string testName)
 		{
-			if (!File.Exists (TestProjectPath))
-				throw new InvalidOperationException ($"Test project '{TestProjectPath}' not found");
+			string tempRoot = Path.Combine (TestOutputDir, $"{testName}.build", XABuildPaths.Configuration);
+			string temporaryProjectPath = Path.Combine (tempRoot, "project");
+			CopyRecursively (TestProjectRootDirectory, temporaryProjectPath);
+			CopyRecursively (CommonSampleLibraryRootDirectory, Path.Combine (tempRoot, CommonSampleLibraryName));
+			return temporaryProjectPath;
 		}
 
-		void ResetProject (string testName)
+		void CopyRecursively (string fromDir, string toDir)
 		{
-			RemoveDir (TestProjectObjPath);
-			RemoveDir (TestProjectBinPath);
+			if (String.IsNullOrEmpty (fromDir))
+				throw new ArgumentException ($"{nameof (fromDir)} is must have a non-empty value");
+			if (String.IsNullOrEmpty (toDir))
+				throw new ArgumentException ($"{nameof (toDir)} is must have a non-empty value");
 
-			string outputDir = GetTestOutputDir (testName);
-			if (Directory.Exists (outputDir))
-				Directory.Delete (outputDir, true);
+			var fdi = new DirectoryInfo (fromDir);
+			if (!fdi.Exists)
+				throw new InvalidOperationException ($"Source directory '{fromDir}' does not exist");
 
-			RemoveLogs (testName);
-		}
+			if (Directory.Exists (toDir))
+				Directory.Delete (toDir, true);
 
-		void RemoveLogs (string testName)
-		{
-			RemoveFile (GetTestLogName (testName));
-			foreach (string log in log_files) {
-				RemoveFile (log);
+			foreach (FileSystemInfo fsi in fdi.EnumerateFileSystemInfos ("*", SearchOption.TopDirectoryOnly)) {
+				if (fsi is FileInfo finfo)
+					CopyFile (fsi.FullName, Path.Combine (toDir, finfo.Name));
+				else
+					CopyRecursively (fsi.FullName, Path.Combine (toDir, fsi.Name));
 			}
+		}
 
-			foreach (string logFile in Directory.EnumerateFiles (TestProjectRootDirectory, $"{ProjectName}.*.log")) {
-				File.Delete (logFile);
-			}
+		void CopyFile (string from, string to)
+		{
+			string dir = Path.GetDirectoryName (to);
+			if (!Directory.Exists (dir))
+				Directory.CreateDirectory (dir);
```

Comment:
You don't need to wrap this in `Directory.Exists()`; `Directory.CreateDirectory()` is a no-op if it already exists.

Reply:
That's true, but... ditto :)